packages/utility-features/utilities/src/isReactNode.ts

```diff
@@ -3,12 +3,31 @@
  * Licensed under MIT License. See LICENSE in root directory for more information.
  */
 
-// Type guard for checking valid react node to use in render
-import type { ReactNode } from "react";
 import React from "react";
 import { isObject } from "./type-narrowing";
 
-export function isReactNode(node: unknown): node is ReactNode {
+// ref: https://changelog.com/posts/the-react-reactnode-type-is-a-black-hole
+
+export type StrictReactFragment =
+  | {
+      key?: string | number | null;
+      ref?: null;
+      props?: {
+        children?: StrictReactNode;
+      };
+    };
+
+export type StrictReactNode =
+  | React.ReactElement
+  | React.ReactText
+  | StrictReactFragment
+  | React.ReactPortal
+  | Iterable<StrictReactNode>
+  | boolean
+  | null
+  | undefined;
```

Comment:
Even if react 18 now permits rendering `undefined`, should we internally advocate that? WDYT?

Reply:
Probably a good idea to remove, will try it.